sync/sync.go

```diff
@@ -494,6 +548,9 @@ func (s *Synchronizer) fetchAndStorePending(ctx context.Context) error {
 		return err
 	}
 
+	// send the pending transactions to the feed
+	s.pendingTxsFeed.Send(pendingBlock.Transactions)
```

Comment:
I don't think this is the correct place to send the pending transactions, please move it below line 618. This is because polling pending block can return the same block multiple times and we don't want to send the same set of transactions to the user multiple times. Only in the `StorePeniding()` do we determine to update the pending block hence base on that decision we should send the pending transactions.